crates/astria-sequencer/src/transaction/unsigned.rs

```diff
@@ -0,0 +1,122 @@
+use astria_proto::sequencer::v1::{
+    unsigned_transaction::Value::AccountsTransaction as ProtoAccountsTransaction,
+    UnsignedTransaction as ProtoUnsignedTransaction,
+};
+use prost::Message as _;
+use serde::{
+    Deserialize,
+    Serialize,
+};
+
+use crate::{
+    accounts::transaction::Transaction as AccountsTransaction,
+    crypto::SigningKey,
+    hash,
+    transaction::signed::Transaction as SignedTransaction,
+};
+
+/// Represents an unsigned sequencer chain transaction.
+/// This type wraps all the different module-specific transactions.
+/// If a new transaction type is added, it should be added to this enum.
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[non_exhaustive]
+pub(crate) enum Transaction {
+    AccountsTransaction(AccountsTransaction),
+}
+
+impl Transaction {
+    /// Attempts to encode the unsigned transaction into bytes.
+    #[must_use]
+    pub(crate) fn to_vec(&self) -> Vec<u8> {
```

Comment:
I find this naming very confusing. Please just make this `fn to_proto -> ProtoUnsignedTransaction` and let the conversion to protobuf be explicit.